admin_panel/src/commonMain/kotlin/net/thechance/mena/admin_panel/presentation/screen/dukan_managements/DukanManagementViewmodel.kt

```diff
@@ -0,0 +1,159 @@
+class DukanManagementViewmodel(
+    @Provided
+    private val dukanRepository: DukanRepository,
+    private val dispatcher: CoroutineDispatcher = Dispatchers.IO
+) : BaseViewModel<DukanManagementScreenState, DukanManagementEffect>(
+    DukanManagementScreenState()
+), DukanManagementInteractionListener {
+    private var dukans: List<Dukan> = emptyList()
+    private var searchJob: Job? = null
+
+    init {
+        getDukans()
+    }
+
+    override fun onSearchQueryChange(query: String) {
+        if (query == currentState.query) return
+
+        updateState { it.copy(query = query) }
+        searchJob?.cancel()
+
+        searchJob = viewModelScope.launch {
+            delay(SEARCH_DEBOUNCE_DELAY)
+            updateState { it.copy(pageInfo = it.pageInfo.copy(page = 0)) }
+            getDukans()
+        }
+    }
+
+    override fun onClearQueryClicked() {
+        updateState {
+            it.copy(query = "")
+        }
+        getDukans()
+    }
+
+    override fun onRetryClicked() {
+        getDukans()
+    }
+
+    override fun onPageChanged(page: Int) {
+        updateState { it.copy(pageInfo = it.pageInfo.copy(page = page)) }
+        getDukans()
+    }
+
+    override fun onViewDetailsClicked(dukanId: Uuid) {
+        val dukan = dukans.find { it.id == dukanId } ?: return
+        dukanRepository.storeDukanDetails(dukan)
+        sendEffect(DukanManagementEffect.NavigateToDukanDetails(dukanId))
+    }
+
+    override fun onSortClicked(type: DukansSortType) {
+        val newDirection = if (currentState.sort.type == type) {
+            toggleSortDirection(currentState.sort.direction)
+        } else {
+            SortDirection.ASC
+        }
+        updateState {
+            it.copy(
+                sort = DukanManagementScreenState.SortState(
+                    type = type,
+                    direction = newDirection
+                ),
+                pageInfo = it.pageInfo.copy(page = 0)
+            )
+        }
+        getDukans()
+    }
+
+    override fun mapError(throwable: Throwable): ErrorState {
+        return when (throwable) {
+            is NoInternetException -> ErrorState.NoInternet
+            else -> ErrorState.UnknownError
+        }
+    }
+
+    private fun getDukansQueryParams(): DukanQueryParams {
+        val trimmedQuery = currentState.query
+            .trim().replace(Regex("\\s+"), " ").ifBlank { null }
+        return DukanQueryParams(
+            searchInput = trimmedQuery,
+            sortType = currentState.sort.type,
+            sortDirection = currentState.sort.direction,
+            page = currentState.pageInfo.page,
+            size = PAGE_SIZE,
+            status = Dukan.Status.APPROVED,
+            activationStatus = null,
+        )
+    }
+
+    private fun getDukans() {
+        tryToExecute(
+            callee = { dukanRepository.getDukans(getDukansQueryParams()) },
+            onSuccess = ::onSuccessGetDukans,
+            onError = ::onFailureGetDukans,
```

Comment:
Just for consistency

```suggestion
            onSuccess = ::onGetDukansSuccess,
            onError = ::onGetDukansError,
```